torchtune/modules/attention.py

```diff
@@ -3,14 +3,14 @@
 #
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
-
+import copy
 from typing import Optional
 
 from torch import nn, Tensor
 from torchtune.modules.kv_cache import KVCache
 
 
-class CausalSelfAttention(nn.Module):
+class GroupedQueryAttention(nn.Module):
```

Comment:
I didn't take changing the name lightly as it creates more work for myself. I think CSA after my changes is too misleading to keep though as it's not self attention specific or causal. The reason I choose GQA is because that's what's specific about our implementation of the module. spda does most of the work for regular MHA, but our module adds extra logic to support grouping. This extra work was also something I don't think we should duplicate in the codebase as it's very important to keep it accurate and well tested.

That said, I'm okay with the name MultiHeadedAttention as well but I think it does less to convey what separates this from spda